### ISSUE TITLE
Replace formatting parser

### ISSUE THREAD

**Issue by DaRacci** [Zrips/CMILib#11]
The colour and formatting parser should be replaced with [MiniMessage](https://docs.adventure.kyori.net/minimessage/) which is cleaner, faster, and more user friendly. Most if not all plugins especially new are moving to minimessage which is now officially supported by paper and used as the message formatting standard.

**Comment by DaRacci** [comment]
Will probably require a user warning, auto convertor or some form of user notice about the change so end users are not confused to why their lang files are no longer supported in the current format.

**Comment by mrfloris** [comment]
Just a side note, 

While supported by Paper, CMI does currently support Spigot, which does not have MiniMsg, meaning CMI would have to still fall back. 

But yes, it would be nice since MiniMsg is baked into Paper, that at some point it could clean up CMI and other zrips plugins by using minimsg instead. But I do feel this has to still work on Spigot, or Spigot support has to be dropped 100%  (which I don't mind, personally).

**Comment by DaRacci** [comment]
> Just a side note, 
> 
> While supported by Paper, CMI does currently support Spigot, which does not have MiniMsg, meaning CMI would have to still fall back. 
> 
> But yes, it would be nice since MiniMsg is baked into Paper, that at some point it could clean up CMI and other zrips plugins by using minimsg instead. But I do feel this has to still work on Spigot, or Spigot support has to be dropped 100%  (which I don't mind, personally).

I agree that dropping spigot is something that can benefit almost all plugins but we wouldn't be able to drop support since spigot requires you to support plain spigot to be listed on spigotmc still I believe.

An alternative would be using a solution which can download libraries at runtime such as the library I maintain slimjar, it would download and mount the adventure libraries if they aren't present in the class path already, meaning on spigot it will download them, on paper it would just work.

**Comment by mrfloris** [comment]
CMILib is already a library, people complain about having a dependency, I don't think offering more will help. I am even trying to get the economy stuff inside cmilib just so we don't need injector, or some custom compile of vault. 
(/offtopic)

**Comment by DaRacci** [comment]
> CMILib is already a library, people complain about having a dependency, I don't think offering more will help. I am even trying to get the economy stuff inside cmilib just so we don't need injector, or some custom compile of vault. 
> (/offtopic)

The slimjar library isn't a plugin itself, it needs to be shaded into the plugin, it doesn't have any dependencies and has a very small footprint

**Comment by DaRacci** [comment]
> CMILib is already a library, people complain about having a dependency, I don't think offering more will help. I am even trying to get the economy stuff inside cmilib just so we don't need injector, or some custom compile of vault. 
> (/offtopic)

The slimjar library isn't a plugin itself, it needs to be shaded into the plugin, it doesn't have any dependencies and has a very small footprint
